src/jit/importer.cpp

```diff
@@ -4179,21 +4339,207 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
             result = NI_System_Collections_Generic_EqualityComparer_get_Default;
         }
     }
-
 #ifdef FEATURE_HW_INTRINSICS
-#if defined(_TARGET_XARCH_)
-    if (strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0)
+    else if (strncmp(namespaceName, "System.Runtime.Intrinsics", 25) == 0)
     {
-        result = HWIntrinsicInfo::lookupId(className, methodName);
-    }
+        namespaceName += 25;
+
+        if (namespaceName[0] == '\0')
+        {
+            if (strncmp(className, "Vector", 6) == 0)
+            {
+                className += 6;
+
+#if defined(_TARGET_ARM64_)
+                if (strncmp(className, "64", 2) == 0)
+                {
+                    className += 2;
+
+                    if (strcmp(className, "`1") == 0)
+                    {
+                        if (strncmp(methodName, "As", 2) == 0)
+                        {
+                            methodName += 2;
+
+                            // Vector64_As, Vector64_AsDouble, Vector64_AsInt64, and Vector64_AsUInt64
```

Comment:
@CarolEidt, I think this is worth another review pass, as I needed to change a few things.

Namely, taking care of this in the importer (https://github.com/dotnet/coreclr/pull/21198/files#diff-edc46b80f57431489a82311948a8234dR4166) only works when both the source and destination type are the same `TYP_SIMD*`.

Since we don't support `TYP_SIMD32` for ARM64 right now, we can't fold away the call. The same goes for `TYP_SIMD8` on x86. Additionally, for `TYP_SIMD8`, we can't fold `AsDouble`, `AsInt64` or `AsUInt64` (and can't easily fold away `As<T>`).

Given that these are special "primtive" types (with corresponding VM support to match the ABI), I think we would want to track these properly (eventually), even if we don't necessarily support the type fully (via the other intrinsics, codegen, etc).

Reply:
> for TYP_SIMD8, we can't fold AsDouble, AsInt64 or AsUInt64

To elaborate on this, ARM64 supports `1D` for vector types on certain vector instructions, but not all. So the JIT needs to know when it is okay to emit the `V.1D` encoding and when it needs to emit a regular scalar instruction (which isn't something it handles today) and given that `__m64` has different ABI requirements from plain `double`, it is likely important for us to track this appropriately, all the way through the JIT (the same goes for `__m128`, `__m256`, and any future types that fall into a similar category, such as `__m512` or `int128`)

Reply:
> Since we don't support TYP_SIMD32 for ARM64 right now, we can't fold away the call. The same goes for TYP_SIMD8 on x86. Additionally, for TYP_SIMD8, we can't fold AsDouble, AsInt64 or AsUInt64 (and can't easily fold away As<T>).

For hardware not supported vector types and base types, JIT should just reject expanding the intrinsic in VectorXXX classes and use software fallback instead. For example, on X86, all the methods in `Vector64` should not be expanded. That is why we have software fallback for platform-agnostic APIs.

Reply:
@fiigii, I was not indicating we should have codegen support. Rather, I was indicating that these are ABI defined types with special semantics.

For example, `Vector64<T>` is defined to have 8-byte packing, even on 32-bit (where `double` and `int64`/`uint64` only have 4 byte packing).

Because these are ABI types and because they may be used as opaque interchange types eventually, having them tracked properly through the JIT is probably important (especially where supporting just the concept of the opaque type is easy), even if we dont support generating any actual code that does anything other than copy the bits around appropriately.

Reply:
So, I was indicating that we should probably support the minimum here, which is recognizing `Vector64<T>` as `TYP_SIMD8`, which is required for ARM/ARM64 and isnt complete today (due to double/int64/uint64). 

Additionally tracking it as `TYP_SIMD8` on x86 should also be possible and should make it easier to handle as an ABI interchange type.

Reply:
@tannergooding Thanks for the explanation. Tracking `Vector64<T>` as TYP_SIMD8 on x86 looks really tricky but useful to implement Vector2 operations in HW intrinsic.

Reply:
If I understand correctly, you're suggesting that they should all be "supported" and recognized types on all platforms, which I think makes sense.
However, you say that to fold them they have to be the same SIMD type. It would seem to me that we could also "fold" them by wrapping the value in an `OBJ` node with the right handle. But moving toward clean "handle-free" SIMD types is probably a better direction.